helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/PartitionMovementConstraint.java

```diff
@@ -44,8 +44,6 @@
   // This factor indicates the default score that is evaluated if only partition allocation matches
   // (states are different).
```

Comment:
Let's revise this description. Confusing - partition allocation matches what?
states are different -> do you mean states are not factored in this calculation?

Reply:
Feel free to open new issues for this. I also think it is confusing. But it is not related to this PR.